go-controller/hack/update-codegen.sh

```diff
@@ -0,0 +1,33 @@
+#!/usr/bin/env bash
+
+set -o errexit
+set -o nounset
+set -o pipefail
+
+SCRIPT_ROOT="$(dirname "${BASH_SOURCE[0]}")/.."
+
+if  ! ( command -v controller-gen > /dev/null ); then
+  # Need to take an unreleased version of controller-tools
+  # because of controller-tools bug 302
+  echo "controller-gen not found, installing sigs.k8s.io/controller-tools@83f6193..."
+  olddir="${PWD}"
+  builddir="$(mktemp -d)"
+  cd "${builddir}"
+  GO111MODULE=on go get -u sigs.k8s.io/controller-tools/cmd/controller-gen@83f6193
+  cd "${olddir}"
+  if [[ "${builddir}" == /tmp/* ]]; then #paranoia
+      rm -rf "${builddir}"
+  fi
+fi
+
+#  github.com/openshift/openshift-network-operator/pkg/generated github.com/openshift/cluster-network-operator/pkg/apis \
+#  "network:v1" \
+#  --go-header-file "${SCRIPT_ROOT}/hack/custom-boilerplate.go.txt"
+
+echo "Generating deepcopy funcs"
+deepcopy-gen --input-dirs github.com/ovn-org/ovn-kubernetes/go-controller/pkg/crd/egressfirewall/v1 -O zz_generated.deepcopy --bounding-dirs github.com/ovn-org/ovn-kubernetes/go-controller/pkg/crd
+
+
+echo "Generating CRDs"
```

Comment:
Hm... I thought for the OpenShift types we were writing CRDs and then generating golang structs from that? Am I wrong or are we doing it backwards here, and if so, why?

Reply:
@danwinship I am pretty sure this is the way this is generated, I took that section directly from the cluster-network-operator. Do you know of a way to generate the golang structs from a CRD?

Reply:
I think we do it the other way around in `openshift/api`... @abhat ?

Reply:
Yes, we write the types.go first and then generate the CRDs from them using controller-gen in `openshift/api`